### PR TITLE
Reorg text-sharing shortcodes before we add more

### PR DESCRIPTION
**Pull request by chalin**
- Suggested prep for #2852
- There are no changes to the generated site files
  ```console
  $ npm run build    
  ...
  $ (cd public && git diff -I '^<meta \w+="[^"]*(date|time)' -I 'Last modified' -- . ':(exclude)*.xml') # no output 
  $
  ```

/cc @svrnm

### CODE REVIEW

File: layouts/shortcodes/docs/instrumentation/lang-index-head.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{{ $data := index $.Site.Data.instrumentation.languages (.Get 0) }}
```
Comment (chalin) [inline layouts/shortcodes/docs/instrumentation/lang-index-head.md:1]: Maybe this should be renamed to `lang-index-intro.md`? WDYT @svrnm?

Comment (chalin) [reply layouts/shortcodes/docs/instrumentation/lang-index-head.md:1]: Yeah ... I prefer "intro". The term "head" makes me think that it includes content in `<head>`. Let me make the change.

File: layouts/shortcodes/docs/instrumentation/lang-index-intro.md
Comment (svrnm) [inline layouts/shortcodes/docs/instrumentation/lang-index-intro.md:1]: maybe _index-intro or even _index/intro (and then later manual/intro, automatic/intro)? so `paganeme/section` would be the pattern (I assume that we will have more than one of these eventually per page)

**Comment by svrnm** [comment]
thank you @chalin 

I agree that we should standardize this part first before proceeding with #2825.

**Review by ferd — APPROVED** [review]

**Review by arielvalentin — APPROVED** [review]
👍🏼 on behalf of @open-telemetry/ruby-approvers

File: layouts/shortcodes/docs/instrumentation/lang-index-intro.md
Comment (chalin) [reply layouts/shortcodes/docs/instrumentation/lang-index-intro.md:1]: We just discussed this out-of-band with @svrnm, and we'll go for `_index-intro.md` for now. We'll refactor further at a later time as needed.

Comment (chalin) [reply layouts/shortcodes/docs/instrumentation/lang-index-intro.md:1]: (Actually, I'm going to drop the leading `_`. It doesn't add much value to distinguish between `index` and `_index` in the case of the shortcode.)

**Comment by svrnm** [comment]
lgtm @chalin, thanks!

**Comment by chalin** [comment]
Given that there are no changes to the generated site files, and this is just a reorg, I'll merge this now.

**Pull request merged**